src/inmanta/agent/forking_executor.py

```diff
@@ -175,15 +200,27 @@ def __init__(
         sources: list[inmanta.loader.ModuleSource],
         environment: uuid.UUID,
         uri: str,
+        venv_checkup_interval: typing.Optional[int] = None,
     ):
+        """
+        :param venv_checkup_interval: The time interval after which the virtual environment must be touched. Only used for
+            testing. The default value is set to None as it should not be used except for testing purposes. It can be
+            overridden to speed up the tests
+        """
         self.venv_path = venv_path
         self.storage_folder = storage_folder
         self.gid = session_gid
         self.sources = sources
         self.environment = environment
         self.uri = uri
+        self._venv_checkup_interval = venv_checkup_interval
 
     async def call(self, context: ExecutorContext) -> typing.Sequence[inmanta.loader.ModuleSource]:
+        assert context.server.timer_venv_checkup is None, "InitCommand should be only called once!"
+
+        interval = self._venv_checkup_interval or 60
+        context.server.start_timer_venv_checkup(interval)
```

Comment:
`self.context.env` is not set yet, if any await call is added between here and line 243, it will give an assertion failure